include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -2517,6 +2731,9 @@ class binary_reader
     /// whether we can assume little endianness
     const bool is_little_endian = little_endianness();
 
+    /// input format
+    input_format_t input_format;
```

Comment:
I am happy to set a default format here - but which one should be the default? `input_format_t::json`?

this value is initialized in the constructor
https://github.com/nlohmann/json/pull/3336/files#diff-b6ced2364e57f63f128546998c2d87f87c590245fd9e8713af2d2c762f5b9739R102-R103

Reply:
If there is no clear "right" default, then whatever the 0 value is.  It's currently initialized in the constructor, but if someone adds a new constructor later and doesn't set that value, then it will be uninitialized and thus non-deterministic, resulting in hard to reproduce errors.  Better to have a deterministic value to be safe.